src/client/datascience/notebook/notebookEditorProvider.ts

```diff
@@ -184,7 +184,6 @@ export class NotebookEditorProvider implements INotebookEditorProvider {
             return;
         }
         this.trackedVSCodeNotebookEditors.add(editor);
-        this.disposables.push(editor.onDidDispose(() => this.onDidDisposeVSCodeNotebookEditor(editor)));
```

Comment:
I'm trying to make sure that this isn't going to open up cases where we don't dispose. But this was causing some large issues. We remove editors from the editor list either on OnClose or OnDidDispose. But OnDidDispose as per the API is just for the Panel being disposed. This is triggered by something like dragging a notebook editor into a new panel. So just by dragging to a new panel we'll clean up the document and remove it from the editors list. But OnClose / OnOpen is not called on a drag operation like that.

Currently I don't see any cases where we would not get the OnClose event and we just do our cleanup then like we normally would. onDidCloseNotebookDocument calls the same disposeResourceRelatedToNotebookEditor that this function did.

Also this was only on changing the active document. I would think that if we needed to actually catch all disposes than this would need to be onDidOpenNotebookDocument.